--- src/libscotchmetis/metis_graph_part_dual_f.c.org	2023-02-05 11:59:23
+++ src/libscotchmetis/metis_graph_part_dual_f.c	2025-05-07 11:53:22
@@ -55,6 +55,8 @@
 #include "scotch.h"
 #include "metis.h"                                /* Our "metis.h" file */
 
+int METIS_PartMeshDual (const SCOTCH_Num * const, const SCOTCH_Num * const, const SCOTCH_Num * const, const SCOTCH_Num * const, const SCOTCH_Num * const, const SCOTCH_Num * const, const SCOTCH_Num * const, const SCOTCH_Num * const, const double * const, const SCOTCH_Num * const, SCOTCH_Num * const, SCOTCH_Num * const, SCOTCH_Num * const);
+
 /**************************************/
 /*                                    */
 /* These routines are the Fortran API */
