From 10cce1222971ff8e26c3aa6bbe095a814de7afe1 Mon Sep 17 00:00:00 2001
From: David Andrs <andrsd@gmail.com>
Date: Sun, 6 Aug 2023 13:47:21 -0600
Subject: [PATCH 2/5] Create a shared version of the MUMPS library.

---
 Makefile     |  8 +++----
 src/Makefile | 61 ++++++++++++++++++++++++++++++++++++----------------
 2 files changed, 47 insertions(+), 22 deletions(-)

diff --git a/Makefile b/Makefile
index f1edac9..67c22b8 100644
--- a/Makefile
+++ b/Makefile
@@ -14,19 +14,19 @@ all: prerequisites
 	cd examples; $(MAKE) all
 
 s: prerequisites
-	cd src; $(MAKE) s
+	cd src; $(MAKE) s ssh
 	cd examples; $(MAKE) s
 
 d: prerequisites
-	cd src; $(MAKE) d
+	cd src; $(MAKE) d dsh
 	cd examples; $(MAKE) d
 
 c: prerequisites
-	cd src; $(MAKE) c
+	cd src; $(MAKE) c csh
 	cd examples; $(MAKE) c
 
 z: prerequisites
-	cd src; $(MAKE) z
+	cd src; $(MAKE) z zsh
 	cd examples; $(MAKE) z
 
 
diff --git a/src/Makefile b/src/Makefile
index 4173b7a..8aab24c 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -9,28 +9,44 @@ incdir = $(topdir)/include
 
 default: d
 
-.PHONY: default all s d c z clean libcommon
+.PHONY: default all s d c z clean libcommon ssh dsh csh zsh libcommonsh
 
-all: $(incdir)/mumps_int_def.h libcommon s d c z
+all: $(incdir)/mumps_int_def.h libcommon s d c z libcommonsh ssh dsh csh zsh
 
 libcommon: $(incdir)/mumps_int_def.h
-	$(MAKE) $(libdir)/libmumps_common$(PLAT)$(LIBEXT)
+	$(MAKE) $(libdir)/libmumps_common$(PLAT).a
 
 s: $(incdir)/mumps_int_def.h libcommon
-	$(MAKE) ARITH=s $(libdir)/libsmumps$(PLAT)$(LIBEXT)
+	$(MAKE) ARITH=s $(libdir)/libsmumps$(PLAT).a
 
 d: $(incdir)/mumps_int_def.h libcommon
-	$(MAKE) ARITH=d $(libdir)/libdmumps$(PLAT)$(LIBEXT)
+	$(MAKE) ARITH=d $(libdir)/libdmumps$(PLAT).a
 
 c: $(incdir)/mumps_int_def.h libcommon
-	$(MAKE) ARITH=c $(libdir)/libcmumps$(PLAT)$(LIBEXT)
+	$(MAKE) ARITH=c $(libdir)/libcmumps$(PLAT).a
 
 z: $(incdir)/mumps_int_def.h libcommon
-	$(MAKE) ARITH=z $(libdir)/libzmumps$(PLAT)$(LIBEXT)
+	$(MAKE) ARITH=z $(libdir)/libzmumps$(PLAT).a
+
+libcommonsh: $(incdir)/mumps_int_def.h
+	$(MAKE) $(libdir)/libmumps_common$(PLAT)$(SHLIB_EXT)
+
+ssh: $(incdir)/mumps_int_def.h libcommonsh
+	$(MAKE) ARITH=s $(libdir)/libsmumps$(PLAT)$(SHLIB_EXT)
+
+dsh: $(incdir)/mumps_int_def.h libcommonsh
+	$(MAKE) ARITH=d $(libdir)/libdmumps$(PLAT)$(SHLIB_EXT)
+
+csh: $(incdir)/mumps_int_def.h libcommonsh
+	$(MAKE) ARITH=c $(libdir)/libcmumps$(PLAT)$(SHLIB_EXT)
+
+zsh: $(incdir)/mumps_int_def.h libcommonsh
+	$(MAKE) ARITH=z $(libdir)/libzmumps$(PLAT)$(SHLIB_EXT)
+
 
 include $(topdir)/Makefile.inc
 
-$(incdir)/mumps_int_def.h: build_mumps_int_def 
+$(incdir)/mumps_int_def.h: build_mumps_int_def
 	./build_mumps_int_def > $(incdir)/mumps_int_def.h
 build_mumps_int_def:build_mumps_int_def.o
 	$(CC) $(OPTC) $(OPTL) build_mumps_int_def.o -o build_mumps_int_def
@@ -192,14 +208,23 @@ OBJS_OTHER = \
         $(ARITH)tools.o\
         $(ARITH)type3_root.o
 
-$(libdir)/libmumps_common$(PLAT)$(LIBEXT):      $(OBJS_COMMON_MOD) $(OBJS_COMMON_OTHER)
-	$(AR)$@ $?
+$(libdir)/libmumps_common$(PLAT).a:      $(OBJS_COMMON_MOD) $(OBJS_COMMON_OTHER)
+	$(AR) $@ $?
 	$(RANLIB) $@
 
-$(libdir)/lib$(ARITH)mumps$(PLAT)$(LIBEXT):    $(OBJS_MOD) $(OBJS_OTHER)
-	$(AR)$@ $?
+$(libdir)/libmumps_common$(PLAT)$(SHLIB_EXT):      $(OBJS_COMMON_MOD) $(OBJS_COMMON_OTHER)
+	$(FC) -shared $^ $(SONAME)libmumps_common$(PLAT)-$(PKG_VERSION)$(SHLIB_EXT) -L$(libdir) $(LORDERINGS) -lpthread $(MPIFLIB) $(MPICLIB) -o $(libdir)/libmumps_common$(PLAT)-$(PKG_VERSION)$(SHLIB_EXT) $(LDFLAGS)
+	ln -sf libmumps_common$(PLAT)-$(PKG_VERSION)$(SHLIB_EXT) $@
+
+$(libdir)/lib$(ARITH)mumps$(PLAT).a:    $(OBJS_MOD) $(OBJS_OTHER)
+	$(AR) $@ $?
 	$(RANLIB) $@
 
+$(libdir)/lib$(ARITH)mumps$(PLAT)$(SHLIB_EXT):    $(OBJS_MOD) $(OBJS_OTHER)
+	$(FC) -shared $^ $(SONAME)lib$(ARITH)mumps$(PLAT)-$(PKG_VERSION)$(SHLIB_EXT) -L$(libdir) -lmumps_common$(PLAT) $(LORDERINGS) $(MPIFLIB) -llapack -lblas $(SCALAP) $(LAPACK) -o $(libdir)/lib$(ARITH)mumps$(PLAT)-$(PKG_VERSION)$(SHLIB_EXT) $(LDFLAGS)
+	ln -sf lib$(ARITH)mumps$(PLAT)-$(PKG_VERSION)$(SHLIB_EXT) $@
+
+
 # Dependencies between modules:
 # i) arithmetic-dependent modules:
 $(ARITH)ana_aux.o:              $(ARITH)mumps_struc_def.o \
@@ -213,7 +238,7 @@ $(ARITH)ana_lr.o:                $(ARITH)lr_core.o\
                                 $(ARITH)lr_stats.o\
                                 lr_common.o\
                                 ana_orderings_wrappers_m.o \
-                                ana_blk_m.o 
+                                ana_blk_m.o
 $(ARITH)fac_asm_master_ELT_m.o: omp_tps_common_m.o \
                                 fac_ibct_data_m.o \
                                 fac_asm_build_sort_index_ELT_m.o \
@@ -225,7 +250,7 @@ $(ARITH)fac_asm_master_ELT_m.o: omp_tps_common_m.o \
                                 $(ARITH)mumps_struc_def.o \
                                 $(ARITH)omp_tps_m.o \
                                 $(ARITH)mumps_comm_buffer.o \
-                                $(ARITH)mumps_load.o 
+                                $(ARITH)mumps_load.o
 
 $(ARITH)fac_asm_master_m.o:     omp_tps_common_m.o \
                                 fac_ibct_data_m.o \
@@ -238,7 +263,7 @@ $(ARITH)fac_asm_master_m.o:     omp_tps_common_m.o \
                                 $(ARITH)mumps_struc_def.o \
                                 $(ARITH)omp_tps_m.o \
                                 $(ARITH)mumps_comm_buffer.o \
-                                $(ARITH)mumps_load.o 
+                                $(ARITH)mumps_load.o
 
 $(ARITH)fac_front_aux.o:        $(ARITH)lr_type.o\
                                 $(ARITH)lr_stats.o\
@@ -411,13 +436,13 @@ $(OBJS_OTHER):$(OBJS_COMMON_MOD) $(OBJS_MOD)
 
 .SUFFIXES: .c .F .o
 .F.o:
-	$(FC) $(OPTF) -I. -I../include $(INCS) $(IORDERINGSF) $(ORDERINGSF) -c $*.F $(OUTF)$*.o
+	$(FC) $(OPTF) -I. -I../include $(INCS) $(IORDERINGSF) $(ORDERINGSF) -fPIC -c $*.F $(OUTF)$*.o
 .c.o:
-	$(CC) $(OPTC) -I../include $(INCS) $(CDEFS) $(IORDERINGSC) $(ORDERINGSC) -c $*.c $(OUTC)$*.o
+	$(CC) $(OPTC) -I../include $(INCS) $(CDEFS) $(IORDERINGSC) $(ORDERINGSC) -fPIC -c $*.c $(OUTC)$*.o
 
 $(ARITH)mumps_c.o:	mumps_c.c
 	$(CC) $(OPTC) -I../include $(INCS) $(CDEFS) -DMUMPS_ARITH=MUMPS_ARITH_$(ARITH) \
-	      $(IORDERINGSC) $(ORDERINGSC) -c mumps_c.c $(OUTC)$@
+	      $(IORDERINGSC) $(ORDERINGSC) -fPIC -c mumps_c.c $(OUTC)$@
 
 clean:
 	$(RM) *.o *.mod $(incdir)/mumps_int_def.h build_mumps_int_def
-- 
2.39.2 (Apple Git-143)

