From 55328ba4130a2180f2da779990b0a669f4e689c5 Mon Sep 17 00:00:00 2001
From: David Andrs <andrsd@gmail.com>
Date: Wed, 20 Aug 2025 10:25:21 -0600
Subject: [PATCH] Adding missing #include

---
 src/single_phase_fluid_properties.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/single_phase_fluid_properties.cpp b/src/single_phase_fluid_properties.cpp
index 905b6fa..6e22655 100644
--- a/src/single_phase_fluid_properties.cpp
+++ b/src/single_phase_fluid_properties.cpp
@@ -7,6 +7,7 @@
 #include "fprops/helium.h"
 #include "fprops/nitrogen.h"
 #include "fprops/exception.h"
+#include <algorithm>
 
 namespace fprops {
 
-- 
2.39.5 (Apple Git-154)

