From 1c1a186875cce7961efe21e5ed3acd01865fc58c Mon Sep 17 00:00:00 2001
From: David Andrs <andrsd@gmail.com>
Date: Wed, 24 Jul 2024 16:13:18 -0600
Subject: [PATCH] Install headers into ${PREFIX}/include/kokkos

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 895cee6a0..f68cc491a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -242,7 +242,7 @@ ELSEIF(HAS_PARENT)
   SET(KOKKOS_HEADER_DIR "include/kokkos")
   SET(KOKKOS_IS_SUBDIRECTORY TRUE)
 ELSE()
-  SET(KOKKOS_HEADER_DIR "${CMAKE_INSTALL_INCLUDEDIR}")
+  SET(KOKKOS_HEADER_DIR "${CMAKE_INSTALL_INCLUDEDIR}/kokkos")
   SET(KOKKOS_IS_SUBDIRECTORY FALSE)
 ENDIF()
 
-- 
2.39.3 (Apple Git-146)

