--- ./ThirdParty/lzma/vtklzma/src/liblzma/common/common.c.org	2023-11-02 13:06:44
+++ ./ThirdParty/lzma/vtklzma/src/liblzma/common/common.c	2023-11-02 13:06:51
@@ -211,7 +211,6 @@
 			|| strm->reserved_ptr2 != NULL
 			|| strm->reserved_ptr3 != NULL
 			|| strm->reserved_ptr4 != NULL
-			|| strm->reserved_int1 != 0
 			|| strm->reserved_int2 != 0
 			|| strm->reserved_int3 != 0
 			|| strm->reserved_int4 != 0
