--- ./Rendering/OpenGL2/CMakeLists.txt.org	2025-03-28 00:59:55
+++ ./Rendering/OpenGL2/CMakeLists.txt	2025-07-29 10:27:37
@@ -448,7 +448,7 @@
         COMPILE_DEFINITIONS "VTK_HAVE_XCURSOR=$<TARGET_EXISTS:X11::Xcursor>"
     )
 elseif (VTK_USE_COCOA)
-  vtk_module_link(VTK::RenderingOpenGL2 PUBLIC "-framework Cocoa")
+  vtk_module_link(VTK::RenderingOpenGL2 PUBLIC "-framework Cocoa -framework OpenGL")
 elseif (VTK_USE_WIN32_OPENGL)
     # vtkWin32OpenGLDXRenderWindow requires D3D.
     vtk_module_find_package(PRIVATE_IF_SHARED
